### PR TITLE
Add support for multiple HMIS IDs (DEV-383)

### PR DESCRIPTION
**Pull request by mikefeldberg**
DEV-383

This PR:
1. adds an `HmisProfile` table to store clients' agency-specific hmis ids. 
2. migrates existing client hmis data to the `HmisProfile` table 
3. updates graphql endpoints

Two follow up steps are required:
1. DEV-384: update frontend to use new fields
2. DEV-524: remove old `hmis_id` from `ClientProfile`, types, queries, and mutations

### CODE REVIEW

File: apps/betterangels-backend/accounts/schema.py [hunk 1]
```diff
@@ -73,12 +73,13 @@ def generate_magic_link(self, info: Info, data: MagicLinkInput) -> MagicLinkResp
...
         with transaction.atomic():
-            client_profile_data: dict = strawberry.asdict(data)
-            user_data = client_profile_data.pop("user") or {}
-
             user = get_current_user(info)
             permission_group = get_user_permission_group(user)
 
+            client_profile_data: dict = strawberry.asdict(data)
+            user_data = client_profile_data.pop("user") or {}
+            hmis_profiles_data = client_profile_data.pop("hmis_profiles", [])
```
Comment (lena) [inline apps/betterangels-backend/accounts/schema.py:81]: just a nitpick, but i expect data to be a `dict`, not a `list`
```suggestion
            hmis_profiles = client_profile_data.pop("hmis_profiles", [])
```

File: apps/betterangels-backend/accounts/tests/test_queries.py [hunk 2]
```diff
@@ -219,7 +242,8 @@ def test_client_profiles_query(self) -> None:
                 }
             }
         """
-        expected_query_count = 3
+        # TODO: why did this go up by 2?
```
Comment (lena) [inline apps/betterangels-backend/accounts/tests/test_queries.py:245]: update?

File: apps/betterangels-backend/accounts/tests/utils.py [hunk 3]
```diff
@@ -60,6 +65,25 @@ def _setup_clients(self) -> None:
...
+        self.client_profile_1_hmis_profile_2 = baker.make(
+            HmisProfile,
+            client_profile_id=self.client_profile_1["id"],
+            hmis_id="A1B2C3PASADENA",
+            agency=HmisAgencyEnum.PASADENA,
+        )
+        self.client_profile_2_hmis_profile = baker.make(
+            HmisProfile,
+            client_profile_id=self.client_profile_2["id"],
+            hmis_id="A1B3C4",
```
Comment (lena) [inline apps/betterangels-backend/accounts/tests/utils.py:83]: make names a lil more scannable/consistent (`PASADENA` here too?)

File: apps/betterangels-backend/accounts/types.py [hunk 4]
```diff
@@ -97,6 +97,18 @@ class LoginInput:
...
+@strawberry_django.type(HmisProfile)
+class HmisProfileType:
+    id: auto
+    hmis_id: auto
+    agency: auto
+
+
+@strawberry_django.input(HmisProfile)
+class HmisProfileInput(HmisProfileType):
+    pass
```
Comment (lena) [inline apps/betterangels-backend/accounts/types.py:109]: nitpick:
```suggestion
    "See parent type"
```

**Review by lena — COMMENTED** [review]
couple comments, looks great overall!

File: apps/betterangels-backend/accounts/tests/utils.py (hunk 3 above)
Comment (mikefeldberg) [reply apps/betterangels-backend/accounts/tests/utils.py:83]: ya, good call

**Review by lena — APPROVED** [review]
👍

**Pull request merged**